src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1346,7 +1346,7 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
             var localSlotManager = new LocalSlotManager(variableSlotAllocatorOpt);
             var optimizations = compilation.Options.OptimizationLevel;
 
-            ILBuilder builder = new ILBuilder(moduleBuilder, localSlotManager, optimizations);
+            ILBuilder builder = new ILBuilder(moduleBuilder, localSlotManager, optimizations, (method as SourceMemberMethodSymbol)?.LocalsAreZeroed ?? true);
```

Comment:
>(method as SourceMemberMethodSymbol)?.LocalsAreZeroed ?? true [](start = 94, length = 61)

It looks like this is not going to cover lambdas and local functions.

Reply:
Our main objective right now is making sure that the attribute works on the method where it's directly applied. In future PRs we'll propagate it to nested lambdas and local functions.